rules/no-reduce.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const methodSelector = require('./utils/method-selector');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const message = 'Array.reduce not allowed';
+
+const PROTOTYPE_SELECTOR = [
+	methodSelector({name: 'call'}),
+	'[callee.object.type="MemberExpression"]',
+	'[callee.object.property.type="Identifier"]',
+	'[callee.object.property.name="reduce"]'
+].join('');
+
+const create = context => {
+	return {
+		[methodSelector({name: 'reduce'})](node) {
```

Comment:
Let's add arguments length check

`methodSelector({name: 'reduce', min:1, max: 2})`

Reply:
added this case in tests too!

Reply:
I always do this tests when I use `methodSelector`, https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/test/no-fn-reference-in-iterator.js#L67-L80